feat(systemd): Add support for unmasking & masking units

fiftydinar opened the pull request.

fiftydinar committed:
feat(systemd): Add support for masking services

fiftydinar committed:
Use unit term in README

fiftydinar committed:
Fix README grammar

fiftydinar committed:
Note about unmasking

fiftydinar committed:
Fix typo in README

xynydev commented on modules/systemd/README.md:
```diff
@@ -1,6 +1,6 @@
 # `systemd` Module for Startingpoint
 
-The `systemd` module streamlines the management of systemd units during image building. Units are divided into `system` and `user` categories, with `system` units managed directly using `systemctl` and `user` units using `systemctl --user`. You can specify which units to enable or disable under each category.
+The `systemd` module streamlines the management of systemd units during image building. Units are divided into `system` and `user` categories, with `system` units managed directly using `systemctl` and `user` units using `systemctl --user`. You can specify which units to enable, disable or mask under each category. To unmask the unit, simply remove it from the mask list.
```
> To unmask the unit, simply remove it from the mask list.

What if the base image already has a masked unit? I'd think that not doing anything about it doesn't automatically unmask it. 

Also, I was not familiar with masking before this, so I'm wondering if we should explain what it is. Or maybe just link to https://fedoramagazine.org/systemd-masking-units/ or something...

fiftydinar replied:
> > To unmask the unit, simply remove it from the mask list.
> 
> What if the base image already has a masked unit? I'd think that not doing anything about it doesn't automatically unmask it.
> 
> Also, I was not familiar with masking before this, so I'm wondering if we should explain what it is. Or maybe just link to https://fedoramagazine.org/systemd-masking-units/ or something...

Base images don't have masked units unless users explicitly mask them.
But I can include unmask option if desired.

It is useful for disabling Power Profiles Daemon as an practical example (when wanting to use System-76 Power). Also useful for TLP. That's in their official guide. There may also be some edge usecases where disabling a service you don't want is not enough.

I explained shortly what masking is (does not run for the user, on any circumstances), but I can also put a link for more details as an less known option.

xynydev replied:
> Base images don't have masked units unless users explicitly mask them.

We can't assume what base images people use. It would be totally possible that someone uses a image built using startingpoint and this module as their base image.

fiftydinar replied:
> > Base images don't have masked units unless users explicitly mask them.
> 
> We can't assume what base images people use. It would be totally possible that someone uses a image built using startingpoint and this module as their base image.

I will add unmasking option then.

fiftydinar replied:
Addressed

fiftydinar committed:
Add unmask & more info about systemctl commands

Address:

https://github.com/ublue-os/bling/pull/88#discussion_r1436877565

https://github.com/ublue-os/bling/pull/88#discussion_r1436917124

xynydev committed:
docs: clarify purpose of unmasking

fiftydinar committed:
docs: Fix systemd module README typo
